src/main/java/com/ryandw11/structure/structure/StructureHandler.java

```diff
@@ -166,6 +168,25 @@ public boolean validDistance(Structure struct, Location location) {
         return struct.getStructureLocation().getDistanceFromOthers() < closest;
     }
 
+    public boolean validSameDistance(Structure struct, Location location) {
+
+        synchronized (spawnedStructures) {
+            double closest = Double.MAX_VALUE;
+            for (Map.Entry<Pair<Location, Long>, Structure> entry : spawnedStructures.entrySet()) {
+
+                if (entry.getKey().getLeft().getWorld() != location.getWorld()) continue;
+
+                if (entry.getKey().getLeft().distance(location) < closest)
+
+                    if (Objects.equals(entry.getValue().getName(), struct.getName())) {
+                        closest = entry.getKey().getLeft().distance(location);
+                    }
+            }
```

Comment:
Maybe keep the return early pattern if we're already using it before to be consistent.
```suggestion
            for (Map.Entry<Pair<Location, Long>, Structure> entry : spawnedStructures.entrySet()) {

                if (entry.getKey().getLeft().getWorld() != location.getWorld()) continue;
                
                if (!Objects.equals(entry.getValue().getName(), struct.getName())) continue;

                if (entry.getKey().getLeft().distance(location) < closest){
                        closest = entry.getKey().getLeft().distance(location);
                }
            }
```